### DISCUSSION TITLE
Can't see/use latest OpenAI models

### DISCUSSION THREAD

**Discussion by pmojong** [danny-avila/LibreChat#1682] [Help Wanted]
![image](https://github.com/danny-avila/LibreChat/assets/103774247/804a57a9-b77c-4508-81a4-f408e9fb472f)
Can't see anything past the GPT-4 (8K) version, but I'm rather certain I do have API access to GPT-4-turbo preview, and the latest release 0125-preview. As seen in the OpenAI playground. 
![image](https://github.com/danny-avila/LibreChat/assets/103774247/7c8ea7fd-0854-404d-86bc-ee0efe0d600d)
I have the newest version of Librechat, which should support those additional models but not sure why I can't see them. 
![image](https://github.com/danny-avila/LibreChat/assets/103774247/ddc78af2-2fb8-4833-90f2-907582e4e222)
Maybe I'm missing something?

**Comment by danny-avila** [comment]
It depends on your setup.
- if you are using Redis, you need to flush the cache whenever you make changes (or disable it). 
- if you are providing the API key from the frontend (setting env var to `user_provided`), you need to determine your model list from the environment variable `OPENAI_MODELS=gpt-3.5-turbo-1106,gpt-4-turbo-preview,` .etc
- if you are providing your API key in the .env file, you should restart your server for the list to be updated from OpenAI's servers.

I will also add the new models to the default list to avoid any confusion when the app does not fetch the list for whatever reason.

**Reply by CodingTobi** [reply]
Hey, 
I had the problem "400 invalid model error" when trying to use the new models. For me the issue was that i pulled the latest version but only updated the OPENAI_MODELS myself in the .env file i already used and did not take into account that there could have been other changes nessesary to the ENV file. 
Solution was to copy the updated .env.example file and adjust the models there and now its working.
For anyone facing the same error.

**Reply by danny-avila** [reply]
Hi @CodingTobi 👋 

Yes it changed, and the expected functionality is being tested automatically and is confirmed working: 
![image](https://github.com/danny-avila/LibreChat/assets/110412045/f0479faa-047f-44b7-abee-863aba253e21)

I can however see your issue. You're supposed to include the list of models without spaces
`OPENAI_MODELS=gpt-4-1106-preview,gpt-4-0125,gpt-4-0125-preview,gpt-4-turbo-preview,gpt-4-turbo`

I'll see about patching this so either/or works.

**Reply by CodingTobi** [reply]
Hey, thanks for the quick reply, that makes sense. Somehow i didn't notice the spaces. Details matter :)
Its working now, thanks for the help.
